pkg/kinflate/app/application.go

```diff
@@ -39,8 +47,102 @@ func New(loader loader.Loader) (Application, error) {
 }
 
 // Resources computes and returns the resources from the manifest.
-func (a *applicationImpl) Resources() []resource.Resource {
+func (a *applicationImpl) Resources() (types.ResourceCollection, error) {
 	// This is where all the fun happens.
 	// From the manifest create the resources.
-	return nil
+	m, err := newManifest(a.loader)
+	if err != nil {
+		return nil, err
+	}
+	a.manifest = *m
+
+	errs := &interror.ManifestErrors{}
+	allResourcesCollection, err := rescourceCollectionFromManifest(a.loader, m)
+	if err != nil {
+		return allResourcesCollection, err
+	}
+
+	for _, pkgPath := range m.Packages {
+		subloader, err := a.loader.New(pkgPath)
+		if err != nil {
+			errs.Append(err)
+			continue
+		}
+		subapp, err := New(subloader)
+		if err != nil {
+			errs.Append(err)
+			continue
+		}
+		subResourcesCollection, err := subapp.Resources()
+		if err != nil {
+			errs.Append(err)
+			continue
+		}
+		types.Merge(allResourcesCollection, subResourcesCollection)
+	}
+
+	patchesRC, err := resourceCollectionFromPaths(a.loader, m.Patches)
+	if err != nil {
+		errs.Append(err)
+		return nil, errs
+	}
+
+	t, err := a.getTransformer(patchesRC)
+	if err != nil {
+		errs.Append(err)
+		return nil, errs
+	}
+	err = t.Transform(types.KObject(allResourcesCollection))
```

Comment:
I am assuming this type-cast will go away once we get rid of types.KObject.

Reply:
True.